### ISSUE TITLE
Running with ShimV2 complains about `shim namespace cannot be empty`

### ISSUE THREAD

**Issue by c3d** [c3d/ociplex#3]
Running with the ShimV2 back-end:

``` shell
% podman --runtime $PWD/run-kata run -it  fedora bash
io.containerd.kata.v2: shim namespace cannot be emptyError: ShimV2 backend "/usr/local/bin/containerd-shim-koto-v2" failed with status code 1
                                                                                                                                             ERRO[0001] Removing container 0d8adb5ac250e9de3ec17d7cb3832692ec83ca6d30344ce3a9a244a900f4ae76 from runtime after creation failed
Error: OCI runtime error: /home/ddd/Work/ociplex/ociplex/run-kata: io.containerd.kata.v2: shim namespace cannot be emptyError: ShimV2 backend "/usr/local/bin/containerd-shim-koto-v2" failed with status code 1

```

Using a temporary interposer that logs what is being done:

``` shell
#!/bin/bash
echo "KOTO: " "$@" >> /tmp/glop
/usr/local/bin/containerd-shim-kata-v2 "$@"
```

The output is the following:

``` shell
KOTO:  start -namespace default -address /run/containerd/containerd.sock -publish-binary /usr/bin/containerd
KOTO:  start -namespace default -address /run/containerd/containerd.sock -publish-binary /usr/bin/containerd
```

So the namespace is set. I think this used to work at some point.

**Comment by c3d** [comment]
The problem is really that the order of arguments passed to the shim is "not standard". Both `containerd.shim.runc.v2` and `containerd.shim.kata.v2` expect the sub-command to be _last_, i.e. instead of `start -namespace default -address /run/containerd/containerd.sock -publish-binary /usr/bin/containerd`, you need to send `-namespace default -address /run/containerd/containerd.sock -publish-binary /usr/bin/containerd start`.

**Comment by c3d** [comment]
This is "fixed" in commit 6a8fb4490b4da5501207902323c44adf3de4c593. The `kata` flag is not well-named, it looks like `runc` behaves the same way.
